config/confighttp/confighttp_test.go

```diff
@@ -63,9 +64,38 @@ func TestAllHTTPClientSettings(t *testing.T) {
 				ReadBufferSize:     1024,
 				WriteBufferSize:    512,
 				CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return next, nil },
+				Compression:        "",
 			},
 			shouldError: false,
 		},
+		{
+			name: "all_valid_settings_with_gzip_compression",
+			settings: HTTPClientSettings{
+				Endpoint: "localhost:1234",
+				TLSSetting: configtls.TLSClientSetting{
+					Insecure: false,
+				},
+				ReadBufferSize:     1024,
+				WriteBufferSize:    512,
+				CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return next, nil },
+				Compression:        "gzip",
+			},
+			shouldError: false,
+		},
+		{
+			name: "error_unsupported_compression",
+			settings: HTTPClientSettings{
+				Endpoint: "localhost:1234",
+				TLSSetting: configtls.TLSClientSetting{
+					Insecure: false,
+				},
+				ReadBufferSize:     1024,
+				WriteBufferSize:    512,
+				CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return next, nil },
+				Compression:        "ggip",
+			},
+			shouldError: true,
+		},
```

Comment:
It looks like the test for no compression was lost in this change, is it still relevant?

Reply:
test for no compression is treated as uncompressed, and it's included in the first test case, "all_valid_settings".